### PR TITLE
Various minor cleanup

### PR DESCRIPTION
**Pull request by dfederm**
Various minor cleanup.

Should review commit-by-commit (this will not be squashed on completion).

### CODE REVIEW

File: src/Build/Definition/ResolvedImport.cs [hunk 1]
```diff
@@ -13,12 +13,6 @@ namespace Microsoft.Build.Evaluation
     /// between a ProjectImportElement and the ProjectRootElement of the
     /// imported project.
     /// </summary>
-    /// <comment>
-    /// This struct is functionally identical to KeyValuePair, but is necessary to avoid
-    /// CA908 warnings (types that in ngen images that will JIT).
-    /// It works because although this is a value type, it is not defined in mscorlib.
-    /// Essentially we would use KeyValuePair except for this technical reason.
-    /// </comment>
```
Comment (Forgind) [inline src/Build/Definition/ResolvedImport.cs]: Why did you delete this?

Comment (dfederm) [reply src/Build/Definition/ResolvedImport.cs]: It's not longer true. `ResolvedImport` is more than just a `KeyValuePair` at this point; in fact it has 5 fields. So if anything it would need to be a value-typed 5-tuple, but that would be a breaking change and not to mention pretty terrible to have to deal with, so overall this comment just doesn't make sense.
